signac/synced_collections/backends/__init__.py

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""This subpackage defines supported backends.
+
+No backends are imported by default. Users should import desired backends as
+needed.
+"""
+__all__ = [""]
```

Comment:
Should it be an empty list, or a list containing an empty string? Is this line needed? If unsure, I vote to delete.

Reply:
Empty list vs empty string are functionally equivalent, but an empty string is necessary to avoid mypy complaining (either that, or explicitly including a type annotation). We could remove this line entirely, but I left it here in anticipation of a discussion of whether we agree that no backends should be imported by default. While I don't want backend classes to be imported at the top level of synced collections (`synced_collections.JSONDict` shouldn't work), I would prefer to be able to do `from synced_collections.backends import JSONDict` rather than `from synced_collections.backends.collection_JSON import JSONDict`. In that case we would populate the `__all__` list appropriately.

Reply:
I don't think we want to trigger loading any backends by default. I would rather the modules be very lightweight and import only what is specifically needed by a user, at the cost of having verbose names.

Reply:
PEP8 says we need `__all__` to be an empty list, not `[""]`. https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces

It's fine if you have to add a type annotation. I'm approving conditional on that change.

Reply:
Sounds good, no backends it is.